src/HIE/Bios/Cradle.hs

```diff
@@ -260,13 +291,57 @@ biosAction wdir bios bios_deps l fp = do
 -- Cabal Cradle
 -- Works for new-build by invoking `v2-repl` does not support components
 -- yet.
+cabalCradleName :: String
```

Comment:
It seems to me that those function ask to be `cradleName :: CradleType -> String` and `isCradleType :: Cradle -> CradleType -> Bool` (so we can check with ```myCradle `isCradleType` StackCradle```)
Or even add `name` as a field to `CradleType` type (although it would suppose more radical changes)

Reply:
I dont like the idea of exposing the Configuration type, if we want to edit that, we are always having breaking changes.
Moreover, it is a lousy pattern match ADT, since we have `StackMulti [(FilePath, CradleConfig)]`. 
However, I like the idea to have a ADT like:
```haskell
data CradleType 
  = Cabal
  | Stack
  | Direct
  ...
  | Other String
```
Which prepares `hie-bios` to be more extensible.